### ISSUE TITLE
Gnome Tracker

### ISSUE THREAD

**Issue by astoff** [abo-abo/swiper#681]
It would be nice to have counsel functions to run [Tracker](https://wiki.gnome.org/Projects/Tracker/) queries: one for full text search as in `counsel-recoll`, and one for searching just file names with `counsel-locate`.

**Comment by abo-abo** [comment]
I've had a short look at `tracker`, it doesn't look very impressive:

    $ tracker-search already
    
    Search term 'already' is a stop word.
    Stop words are common words which may be ignored during the indexing process.
    
    Results:

Is it any good in your experience?

**Comment by astoff** [comment]
Well, I can't really attest to its goodness, but it seems to be a standard component of Gnome and work reasonably well. For example, if I do
`$ touch anyzhing; sleep 5; tracker search anyzhing`
then it actually finds the file.

I am actually more interested in searching files by name, which can surely be done using `track sparql`, although I don't know exactly how. This would be way better than `locate` and its invariably stale db.

**Comment by dalanicolai** [comment]
I know this is a very old-post but still someone arriving here somehow might be interested in [this ulauncher extension](https://github.com/dalanicolai/gnome-tracker-extension) also.

**Comment by basil-conto** [comment]
Does the new `counsel-tracker` command, added in commit 742119ad6fb7ba26e18df2c40a723f36b71287e8, address this feature request?

**Comment by astoff** [comment]
@basil-conto In broad terms, yes. But I noticed two little issues. First, search is case-sensitive. Also, it doesn't seem to find files outside `~/Documents`, `~/Downloads` and other "official" subfolders of the home directory.

Since there are other tools to search files by name (I have been using `(lambda () (interactive) (counsel-fzf nil "~"))` for this and I'm happy with it), wouldn't it make more sense for `counsel-tracker` to focus on the full-text search functionality?

**Comment by basil-conto** [comment]
> wouldn't it make more sense for `counsel-tracker` to focus on the full-text search functionality?

I don't doubt that it would, but I'm not familiar Tracker. Hopefully someone will submit possible enhancements soon.

**Comment by dalanicolai** [comment]
@basil-conto @astoff Just to sharing my perspective. I was new to (spac)Emacs when I proposed this feature. I thought navigating to the right folder was quite cumbersome, while it was much easier in nautilus (see #2484) and this feature was better than nothing. Although I added deep search features to Ulauncher (using recoll, docfetcher or tracker) I don't use deep file search much.

A problem with tracker is that, with default settings, it scans only some particular folders like mentioned already by @astoff and also it only indexes parts of documents (like 10000 words or so).
Of course folders can be added to the index so then you can quickly find any file in any of those folders using `counsel-tracker`, but I think in general if you need deep file search it makes more sense to use recoll. Indeed I think I discovered the counsel-fzf extension, which is a nice alternative, only after this feature proposal but in the end nowadays I mainly use counsel-locate to find files by filename and counsel-search with ripgrep when I do need deep-search.

Anyway, I think what I want to say is that I do not expect enhancements from gnome-tracker when other tools work better for these things. Maybe one day it could be a nice option to use the tracker tag features, but generally there are probably better alternatives for that too, that work on any platform. Of course tracker is very nice for making gnome work great
